kubernetes/schema_resources_migrate_test.go

```diff
@@ -0,0 +1,137 @@
+									"memory": "128Mi",
+								}},
+							}},
+						},
+						map[string]interface{}{
+							"resources": []interface{}{map[string]interface{}{
+								"requests": []interface{}{map[string]interface{}{
+									"cpu":    "200m",
+									"memory": "16Mi",
+								}},
+								"limits": []interface{}{map[string]interface{}{
+									"cpu":    "300m",
+									"memory": "32Mi",
+								}},
+							}},
+						},
+					},
+					"container": []interface{}{
+						map[string]interface{}{
+							"resources": []interface{}{map[string]interface{}{
+								"requests": []interface{}{map[string]interface{}{
+									"cpu":    "500m",
+									"memory": "56Mi",
+								}},
+								"limits": []interface{}{map[string]interface{}{
+									"cpu":    "750m",
+									"memory": "128Mi",
+								}},
+							}},
+						},
+						map[string]interface{}{
+							"resources": []interface{}{map[string]interface{}{
+								"requests": []interface{}{map[string]interface{}{
+									"cpu":    "200m",
+									"memory": "16Mi",
+								}},
+								"limits": []interface{}{map[string]interface{}{
+									"cpu":    "300m",
+									"memory": "32Mi",
+								}},
+							}},
+						},
+					},
+				}},
+			}},
+		}},
+	}
+
+	v1 := map[string]interface{}{
+		"spec": []interface{}{map[string]interface{}{
+			"template": []interface{}{map[string]interface{}{
+				"spec": []interface{}{map[string]interface{}{
+					"init_container": []interface{}{
+						map[string]interface{}{
+							"resources": []interface{}{map[string]interface{}{
+								"requests": map[string]interface{}{
+									"cpu":    "500m",
+									"memory": "56Mi",
+								},
+								"limits": map[string]interface{}{
+									"cpu":    "750m",
+									"memory": "128Mi",
+								},
+							}},
+						},
+						map[string]interface{}{
+							"resources": []interface{}{map[string]interface{}{
+								"requests": map[string]interface{}{
+									"cpu":    "200m",
+									"memory": "16Mi",
+								},
+								"limits": map[string]interface{}{
+									"cpu":    "300m",
+									"memory": "32Mi",
+								},
+							}},
+						},
+					},
+					"container": []interface{}{
+						map[string]interface{}{
+							"resources": []interface{}{map[string]interface{}{
+								"requests": map[string]interface{}{
+									"cpu":    "500m",
+									"memory": "56Mi",
+								},
+								"limits": map[string]interface{}{
+									"cpu":    "750m",
+									"memory": "128Mi",
+								},
+							}},
+						},
+						map[string]interface{}{
+							"resources": []interface{}{map[string]interface{}{
+								"requests": map[string]interface{}{
+									"cpu":    "200m",
+									"memory": "16Mi",
+								},
+								"limits": map[string]interface{}{
+									"cpu":    "300m",
+									"memory": "32Mi",
+								},
+							}},
+						},
+					},
+				}},
+			}},
+		}},
+	}
+
+	actual, _ := upgradeTemplatePodSpecWithResourcesFieldV0(context.Background(), v0, nil)
```

Comment:
This is just a best-practice nit on my part, but if a context doesn't have a deadline or a cancel function, it's best to use [context.TODO()](https://golang.org/pkg/context/#TODO). That makes it easier to find/replace if we do end up using a real context some day.

```suggestion
	actual, _ := upgradeTemplatePodSpecWithResourcesFieldV0(context.TODO(), v0, nil)
```